scripts/submit-analytics.js

```diff
@@ -0,0 +1,208 @@
+      if (/hermes_enabled.*true/.test(podfile)) {
+        jsEngine = "hermes";
+      } else {
+        jsEngine = "jsc";
+      }
+    } catch (err) {
+      doLog(`Cannot read ios/Podfile: ${err}`);
+      jsEngine = "unknown";
+    }
+  }
+  if (context.dependencies && context.dependencies["electron"]) {
+    framework = "electron";
+    frameworkVersion = context.dependencies["electron"];
+  }
+
+  const payload = {
+    webHook: {
+      event: "install",
+      properties: {
+        token: "aab85907a13e1ff44a95be539d9942a9",
+        "JS Analytics Version": 2,
+        distinct_id: identifier,
+        "Anonymized Machine Identifier": identifier,
+        "Anonymized Application ID": sha256(__dirname),
+        Binding: "javascript",
+        Version: context.version,
+        Language: "javascript",
+        Framework: framework,
+        "Framework Version": frameworkVersion,
+        "JavaScript Engine": jsEngine,
+        "Host OS Type": os.platform(),
+        "Host OS Version": os.release(),
+        "Node.js version": process.version,
+      },
+    },
+  };
+
+  return payload;
+}
+
+function isAnalyticsDisabled() {
+  let isDisabled = false;
+
+  // NODE_ENV is commonly used by JavaScript framework
+  if ("NODE_ENV" in process.env) {
+    isDisabled |= process.env["NODE_ENV"] === "production" || process.env["NODE_ENV"] === "test";
+  }
+
+  // If the user has specifically opted-out or if we're running in a CI environment
+  isDisabled |= "REALM_DISABLE_ANALYTICS" in process.env || "CI" in process.env;
+
+  return isDisabled;
+}
+
+/**
+ * Send collected analytics data to Realm's servers over HTTPS
+ * @param  {Object} payload analytics info
+ */
+async function dispatchAnalytics(payload) {
+  const https = require("https");
+
+  return new Promise((resolve, reject) => {
+    const requestUrl = getAnalyticsRequestUrl(payload);
+
+    https
+      .get(requestUrl, (res) => {
+        resolve({
+          statusCode: res.statusCode,
+          statusMessage: res.statusMessage,
+        });
+      })
+      .on("error", (error) => {
+        const message = error && error.message ? error.message : error;
+        const err = new Error(`Failed to dispatch analytics: ${message}`);
+        reject(err);
+      });
+  });
+}
+
+async function submitAnalytics(dryRun) {
+  const wd = process.cwd();
+  const index = wd.indexOf("node_modules");
+  const packageJson = (index === -1 ? wd : wd.slice(0, index)) + "/package.json";
+  const context = require(packageJson);
+  const payload = await fetchPlatformData(context);
+  doLog(`payload: ${JSON.stringify(payload)}`);
+
+  if (dryRun) {
+    doLog("Dry run; will not submit analytics");
+    return;
+  }
+
+  if (isAnalyticsDisabled()) {
+    doLog("Analytics is disabled");
+    return;
+  }
+
+  await Promise.all([dispatchAnalytics(payload)]);
+}
+
+const optionDefinitions = [
+  { name: "dryRun", type: Boolean, multiple: false, description: "If true, don't submit analytics" },
```

Comment:
I'm pretty sure these objects take a default value, making your next couple of lines obsolete.